python/res/enkf/rng_config.py

```diff
@@ -20,19 +20,41 @@
 from ecl.util.enums import RngAlgTypeEnum
 
 from res import ResPrototype
+from res.enkf import ConfigKeys
 
 class RNGConfig(BaseCClass):
 
     TYPE_NAME = "rng_config"
     _alloc        = ResPrototype("void* rng_config_alloc(config_content)", bind=False)
+    _alloc_full   = ResPrototype("void* rng_config_alloc_full(char*, char*, char*)", bind=False)
     _free         = ResPrototype("void rng_config_free(rng_config)")
     _rng_alg_type = ResPrototype("rng_alg_type_enum rng_config_get_type(rng_config)")
     _load_file    = ResPrototype("char* rng_config_get_seed_load_file(rng_config)")
     _store_file   = ResPrototype("char* rng_config_get_seed_store_file(rng_config)")
     _random_seed  = ResPrototype("char* rng_config_get_random_seed(rng_config)")
 
-    def __init__(self, config_content):
-        c_ptr = self._alloc(config_content)
+    def __init__(self, config_content=None, config_dict=None):
+        if config_content and config_dict:
+            raise ValueError('RNGConfig can not be instantiated with both config types')
+
+        if not (config_content or config_dict):
+            raise ValueError('RNGConfig can not be instantiated without any config objects')
+
+        if config_content:
+            c_ptr = self._alloc(config_content)
+
+        elif config_dict:
+            random_seed = None
+            if ConfigKeys.RANDOM_SEED in config_dict:
+                random_seed = config_dict[ConfigKeys.RANDOM_SEED]
+            store_seed = None
+            if not random_seed and ConfigKeys.STORE_SEED in config_dict:
+                store_seed = config_dict[ConfigKeys.STORE_SEED]
+            load_seed = None
+            if not random_seed and ConfigKeys.LOAD_SEED in config_dict:
+                load_seed = config_dict[ConfigKeys.LOAD_SEED]
```

Comment:
This piece of code can be rewritten with 3 one-liners such as
```Python
random_seed = config_dict.get(ConfigKeys.RANDOM_SEED, default=None)
```
The `default=None` part can actually be omitted